client/src/utils/Store.js

```diff
@@ -0,0 +1,162 @@
+import axios from 'axios';
+import clonedeep from 'lodash.clonedeep';
+import baseurl from '@/env';
+import {
+  set,
+  keys,
+  del,
+  get,
+} from 'idb-keyval';
+
+axios.defaults.withCredentials = true;
+export default class API {
+  constructor() {
+    this.plot = '';
+    this.init();
+  }
+
+  /**
+   * initializes cache
+   */
+  init = async () => {
+    const k = await keys();
+    this.models = new Map();
+    k.forEach((key) => {
+      this.models.set(key, get(key));
+    });
+    await Promise.all(this.models);
+  }
+
+  /**
+   * Flattens model to make request params
+   * @param {Object} model model to flatten
+   * @returns {Object} flattened params
+   */
+  flatten = (model) => {
+    const params = {};
+    Object.values(model).forEach((value) => {
+      Object.assign(params, value);
+    });
+    return params;
+  }
```

Comment:
No. I changed it to assign because you're creating a new object every time you using the spread. When you use assign, you're writing it into the same params object. 
params = { ...params, ...value } vs Object.assign(params, value)

Reply:
Imo this is less expressive for no payoff but ok